core-api/src/main/java/com/optimizely/ab/internal/ConditionUtils.java

```diff
@@ -0,0 +1,171 @@
+                        (String)conditionMap.get("match"), conditionMap.get("value")));
+            }
+            else {
+                Map<String, ?> conditionMap = (Map<String, ?>)rawObjectList.get(i);
+                conditions.add(new UserAttribute((String)conditionMap.get("name"), (String)conditionMap.get("type"),
+                        (String)conditionMap.get("match"), conditionMap.get("value")));
+            }
+        }
+
+        Condition condition;
+        switch (operand) {
+            case "and":
+                condition = new AndCondition(conditions);
+                break;
+            case "or":
+                condition = new OrCondition(conditions);
+                break;
+            case "not":
+                condition = new NotCondition(conditions.get(0));
+                break;
+            default:
+                condition = new OrCondition(conditions);
+                break;
+        }
+
+        return condition;
+    }
+
+    static public Condition parseConditions(org.json.JSONArray conditionJson) {
+        List<Condition> conditions = new ArrayList<Condition>();
+        String operand = (String)conditionJson.get(0);
+
+        for (int i = 1; i < conditionJson.length(); i++) {
+            Object obj = conditionJson.get(i);
+            if (obj instanceof org.json.JSONArray) {
+                conditions.add(parseConditions((org.json.JSONArray) conditionJson.get(i)));
+            } else if (obj instanceof String) {
+                conditions.add(new AudienceIdCondition((String)obj));
+            } else {
+                org.json.JSONObject conditionMap = (org.json.JSONObject)obj;
+                String match = null;
+                Object value = null;
+                if (conditionMap.has("match")) {
+                    match = (String) conditionMap.get("match");
+                }
+                if (conditionMap.has("value")) {
+                    value = conditionMap.get("value");
+                }
+                conditions.add(new UserAttribute((String)conditionMap.get("name"), (String)conditionMap.get("type"),
+                        match, value));
+            }
+        }
+
+        Condition condition;
+        switch (operand) {
+            case "and":
+                condition = new AndCondition(conditions);
+                break;
+            case "or":
+                condition = new OrCondition(conditions);
+                break;
+            case "not":
+                condition = new NotCondition(conditions.get(0));
+                break;
+            default:
+                condition = new OrCondition(conditions);
+                break;
+        }
+
+        return condition;
+    }
+
+
+    public static void resolveAudienceIdConditions(@Nonnull ProjectConfig projectConfig,
+                                                   @Nonnull Condition conditions) {
```

Comment:
It's not clear to me what this function does: there's no return value and no obvious side effects. Can you add a JSDoc?